ui/packages/catalog-ui-search/src/main/webapp/js/widgets/openlayers.circle.js

```diff
@@ -64,15 +64,18 @@ Draw.CircleView = Marionette.View.extend({
   },
   setModelFromGeometry: function(geometry) {
     var center = translateFromOpenlayersCoordinate(geometry.getCenter())
+    var rad =
+      geometry.getRadius() *
+      this.map
+        .getView()
+        .getProjection()
+        .getMetersPerUnit()
+
     this.model.set({
-      lat: center[1],
-      lon: center[0],
+      lat: DistanceUtils.coordinateRound(center[1]),
+      lon: DistanceUtils.coordinateRound(center[0]),
       radius: DistanceUtils.getDistanceFromMeters(
-        geometry.getRadius() *
-          this.map
-            .getView()
-            .getProjection()
-            .getMetersPerUnit(),
+        DistanceUtils.coordinateRound(rad),
```

Comment:
Not sure the radius needs this however if it does, then the rounding should be applied after getDistanceFromMeters does the unit of measurement conversion as that could introduce more floating point numbers. An easy test of this would be set a radius of 5 nautical miles, that should convert to 0.002699784017279